components/esp_modem/include/cxx_include/esp_modem_dte.hpp

```diff
@@ -133,9 +139,68 @@ class DTE : public CommandableIf {
     std::shared_ptr<Terminal> primary_term;                 /*!< Reference to the primary terminal (mostly for sending commands) */
     std::shared_ptr<Terminal> secondary_term;               /*!< Secondary terminal for this DTE */
     modem_mode mode;                                        /*!< DTE operation mode */
-    SignalGroup signal;                                     /*!< Event group used to signal request-response operations */
-    command_result result;                                  /*!< Command result of the currently exectuted command */
     std::function<bool(uint8_t *data, size_t len)> on_data; /*!< on data callback for current terminal */
+
+#ifdef CONFIG_ESP_MODEM_USE_INFLATABLE_BUFFER_IF_NEEDED
+    /**
+     * @brief Implements an extra buffer that is used to capture partial reads from underlying terminals
+     * when we run out of the standard buffer
+     */
+    struct extra_buffer {
+        extra_buffer(): buffer(nullptr) {}
+        ~extra_buffer()
+        {
+            delete buffer;
+        }
+        std::vector<uint8_t> *buffer;
```

Comment:
Why use the `vector<>*` instead of `vector<>`?

Reply:
The reason is that I wanted to deallocate the members, which is not under my control if I used just `vector<>`.

The idea is that we're able to grow the extra buffer in exceptional cases and after we do that and process all the data, we should free these blocks.